xmrstak/net/jpsock.cpp

```diff
@@ -411,6 +424,17 @@ bool jpsock::process_pool_job(const opq_json_val* params)
 		return set_socket_error("PARSE error: Job error 4");
 
 	oPoolJob.iWorkLen = iWorkLen;
+
+	{
+		// lock reading of oCurrentJob
+		std::lock_guard<std::mutex> lock(job_mutex);
+		// compare possible non equal length job id's
+		if(std::string(jobid->GetString()) == std::string(oCurrentJob.sJobID))
```

Comment:
So much copying for so little gain... `strcmp(jobid->GetString(), oCurrentJob.sJobID) == 0`

**Please** make sure sJobID is always going to end with `\0` otherwise both versions break.

Reply:
yes both are zero terminated. `oCurrentJob.sJobID` is max 63 characters and the full array (64 char) is zeroed. `GetString()` is giving a C string.

Reply:
This is not sufficient. This is after the compare. Please make sure that sJobID is also zeroed on initialisation.